src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const objDate = {};
+  const oldFormat = fromFormat.slice(0, -1);
+  const newFormat = toFormat.slice(0, -1);
+  const oldDelimeter = fromFormat[3];
+  const newDelimeter = toFormat[3];
+  const arrData = date.split(oldDelimeter);
+  let resDateStr = '';
+
+  for (let i = 0; i < arrData.length; i++) {
+    objDate[oldFormat[i]] = arrData[i];
+  }
+
+  if ('YYYY' in objDate) {
+    objDate.YY = objDate.YYYY.slice(-2);
+  }
+
+  if ('YY' in objDate && newFormat.includes('YYYY')) {
```

Comment:
there may be a case when `oldFormat` contains 'YYYY', you add 'YY' in the previous condition and when the program is on this line, we already have both properties in the objDate. In this case, converting 'YY' to 'YYYY' will be redundant as we already have 'YYYY'